src/libfetchers/github.cc

```diff
@@ -265,14 +265,18 @@ struct GitHubInputScheme : GitArchiveInputScheme
         // FIXME: use regular /archive URLs instead? api.github.com
         // might have stricter rate limits.
         auto host = maybeGetStrAttr(input.attrs, "host").value_or("github.com");
-        auto url = fmt(
-            host == "github.com"
-            ? "https://api.%s/repos/%s/%s/tarball/%s"
-            : "https://%s/api/v3/repos/%s/%s/tarball/%s",
-            host, getStrAttr(input.attrs, "owner"), getStrAttr(input.attrs, "repo"),
+        Headers headers = makeHeadersWithAuthTokens(host);
+        const char* urlFmt;
```

Comment:
Any reason to not opt for a std::string here?

Reply:
afaik, this would create an unnecessary string allocation. Also I cannot use const in this case.

Reply:
I think `std::string_view` should work.

Reply:
You can probably also write this as
```
auto urlFmt =
    host == "github.com"
    ? headers.empty() 
       ? "https://%s/%s/%s/archive/%s.tar.gz" 
       : "https://api.%s/repos/%s/%s/tarball/%s"
    : "https://%s/api/v3/repos/%s/%s/tarball/%s"
```

Reply:
> You can probably also write this as
> 
> ```
> auto urlFmt =
>     host == "github.com"
>     ? headers.empty() 
>        ? "https://%s/%s/%s/archive/%s.tar.gz" 
>        : "https://api.%s/repos/%s/%s/tarball/%s"
>     : "https://%s/api/v3/repos/%s/%s/tarball/%s"
> ```

I find this quite hard to read to be honest.

Reply:
https://github.com/NixOS/nix/pull/7039

Reply:
Closed this pr and move ahead in https://github.com/NixOS/nix/pull/7039